### DISCUSSION TITLE
Getting an error when associating role to a user for one to many relationship

### DISCUSSION THREAD

**Discussion by toxifiedmatrix** [laravel/framework#36082] [Q&A]
I have a one to many relationship between `users` and `roles`. I have three roles `administrator`, `employee` and `client`. I am unable to associate the user with a role. 

Example:
When I am updating a user with a role `administrator`, I am successful in editing and saving the role to `employee` and `client`, but when I click save when the default role value remains unchanged i.e. `administrator`, it throws me an SQLSTATE[22007] error.

Please spare some time to have a look below.

> Illuminate\Database\QueryException
SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect integer value: '[3]' for column `management`.`users`.`role_id` at row 1 (SQL: update `users` set `role_id` = [3], `users`.`updated_at` = 2021-01-29 05:08:12 where `id` = 3)
http://127.0.0.1:8000/backend/management/audience-management/user

![SQLSTATE ERROR](https://user-images.githubusercontent.com/65024122/106234732-5422b100-621f-11eb-9158-8735959f3dd5.gif)

- Database Structure
```
Users
----------
* id
* name
* email
* password
* role_id
* created_at

Roles
----------
* id
* name
* slug
* created_at
```

- User Model
```
public function role() {
    return $this->belongsTo(Role::class);
}
```

- Role Model
```
public function users() {
    return $this->hasMany(User::class);
}
```

- User Controller
```
public $showUserUpdationModal = false;
public $role;
public User $user;

protected $rules = [
    'user.name' => 'required|string|max:255',
    'user.email' => 'required|string|email|max:255',
    'role' => 'required',
];

public function storeUser()
{
    $this->validate();
    $this->validate([
        'user.email' => 'unique:users,email,'.$this->user->id,
    ]);
    $this->user->role()->associate($this->role);
    $this->user->save();
    $this->showUserUpdationModal = false;

    $this->dispatchBrowserEvent('notify', $this->user->name.' Updated Successfully');
    $this->emit('sectionRefresh');
}
```

**Comment by ahinkle** [comment]
Everything looks good - except that you are trying to insert `[3]` as a record into your database; where it's not a valid integer. Usually these things point to the dropdown values in the request.

Take a look at your select dropdown input. What's the value for each role?
